phpBB/phpbb/members/controller/online.php

```diff
@@ -0,0 +1,618 @@
+
+	/**
+	 * @var config
+	 */
+	protected $config;
+
+	/**
+	 * @var driver_interface
+	 */
+	protected $db;
+
+	/**
+	 * @var dispatcher
+	 */
+	protected $dispatcher;
+
+	/**
+	 * @var group_helper
+	 */
+	protected $group_helper;
+
+	/**
+	 * @var helper
+	 */
+	protected $helper;
+
+	/**
+	 * @var language
+	 */
+	protected $language;
+
+	/**
+	 * @var pagination
+	 */
+	protected $pagination;
+
+	/**
+	 * @var request
+	 */
+	protected $request;
+
+	/**
+	 * @var router
+	 */
+	protected $router;
+
+	/**
+	 * @var template
+	 */
+	protected $template;
+
+	/**
+	 * @var user
+	 */
+	protected $user;
+
+	public function __construct(auth $auth, config $config, driver_interface $db, dispatcher $dispatcher, group_helper $group_helper, helper $helper, language $language, pagination $pagination, request $request, router $router, template $template, user $user)
+	{
+		$this->auth				= $auth;
+		$this->config			= $config;
+		$this->db				= $db;
+		$this->dispatcher		= $dispatcher;
+		$this->group_helper		= $group_helper;
+		$this->helper			= $helper;
+		$this->language			= $language;
+		$this->pagination		= $pagination;
+		$this->request			= $request;
+		$this->router			= $router;
+		$this->template			= $template;
+		$this->user				= $user;
+	}
+
+	/**
+	 * Controller for /online route
+	 *
+	 * @return \Symfony\Component\HttpFoundation\Response a Symfony response object
+	 */
+	public function handle()
+	{
+		global $phpbb_root_path, $phpEx;
+
+		// Display a listing of board admins, moderators
+		if (!function_exists('display_user_activity'))
+		{
+			include($phpbb_root_path . 'includes/functions_display.' . $phpEx);
+		}
+
+		// Load language strings
+		$this->language->add_lang('memberlist');
+
+		// Can this user view profiles/memberlist?
+		if (!$this->auth->acl_gets('u_viewprofile', 'a_user', 'a_useradd', 'a_userdel'))
+		{
+			if ($this->user->data['user_id'] != ANONYMOUS)
+			{
+				throw new http_exception(403, 'NO_VIEW_USERS');
+			}
+
+			login_box('', $this->language->lang('LOGIN_EXPLAIN_VIEWONLINE'));
+		}
+
+		// Get and set some variables
+		$mode			= $this->request->variable('mode', '');
+		$session_id		= $this->request->variable('s', '');
+		$start			= $this->request->variable('start', 0);
+		$sort_key		= $this->request->variable('sk', 'b');
+		$sort_dir		= $this->request->variable('sd', 'd');
+		$show_guests	= ($this->config['load_online_guests']) ? $this->request->variable('sg', 0) : 0;
+
+		$sort_key_text = array('a' => $this->language->lang('SORT_USERNAME'), 'b' => $this->language->lang('SORT_JOINED'), 'c' => $this->language->lang('SORT_LOCATION'));
+		$sort_key_sql = array('a' => 'u.username_clean', 'b' => 's.session_time', 'c' => 's.session_page');
+
+		// Sorting and order
+		if (!isset($sort_key_text[$sort_key]))
+		{
+			$sort_key = 'b';
+		}
+
+		$order_by = $sort_key_sql[$sort_key] . ' ' . (($sort_dir == 'a') ? 'ASC' : 'DESC');
+
+		$this->user->update_session_infos();
+
+		$forum_data = $this->get_forum_data();
```

Comment:
`$forum_data` doesn't seem to be used anywhere.

Reply:
Seems to be used in get_location()?

Reply:
Is used in the event

Reply:
And yes, the variable is undefined in the other method